modules/internal/DefaultRectangular.chpl

```diff
@@ -627,11 +621,13 @@ module DefaultRectangular {
     proc dsiHasSingleLocalSubdomain() param return true;
 
     proc dsiLocalSubdomain() {
-      return _newDomain(this);
+      // TODO: this should only return this way if this.locale == here
```

Comment:
Do you intend to fix this before merging? What is the impact of having it wrong as it is now?
It sounds trivial to fix. Is there some reason to hesitate in adding the comparison?